.github/workflows/build.yml

```diff
@@ -0,0 +1,93 @@
+name: Deploy release candidate
+on:
+    push:
+        branches:
+            - main
+    pull_request: # Runs on pull requests to any branch
+env:
+    IS_RELEASE_CANDIDATE: ${{ contains(github.head_ref, 'release--') }}
+    IS_FEATURE: ${{ !contains(github.head_ref, 'release--') }}
+    DFX_VERSION: 0.21.0
+    NODE_VERSION: 20
+jobs:
+    get-tests:
+        if: ${{ contains(github.head_ref, 'release--') }}
+        name: Get tests
+        runs-on: ubuntu-latest
+        outputs:
+            tests: ${{ steps.get_tests.outputs.tests }}
+        steps:
+            - uses: actions/checkout@v4
+            - name: Get all test dirs
+              id: get_tests
+              uses: ./.github/actions/generate_tests
+              with:
+                  directories: |
+                      ./examples
+                      ./tests
+
+    determine-is-manual-release:
+        if: ${{ contains(github.head_ref, 'release--') }}
+        name: Determine if this job is for a manual release
+        runs-on: ubuntu-latest
+        outputs:
+            is_manual_release: ${{ steps.determine_is_manual_release.outputs.is_manual_release }}
+        steps:
+            - uses: actions/checkout@v4
+            - id: determine_is_manual_release
+              run: |
+                  BRANCH_NAME="${{ github.head_ref }}"
+                  RELEASE_VERSION="${BRANCH_NAME:9}"
+                  COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")
+
+                  IS_AUTOMATED_RELEASE="false"
```

Comment:
I'm hoping if we rename everything to `should_release` and `release` that this will become more straight-forward

Reply:
I think it is